specs/commandline/inspect.md

```diff
@@ -0,0 +1,234 @@
+
+# notation inspect
+
+## Description
+
+Use `notation inspect` command to inspect all the signatures associated with signed artifact in a human readable format.
+
+Upon successful execution, both the digest of the signed artifact and the digests of signatures manifest along with their properties associated with the signed artifact are printed in the following format:
+
+```shell
+<registry>/<repository>@<digest>
+└──application/vnd.cncf.notary.signature
+  ├──<digest_of_signature_manifest>
+    ├──<signing algorithm>
+    ├──<signed attributes>
+    ├──<user-defined attributes>
+    ├──<unsigned attributes>
+    ├──<certificates>
+    └──<payload>
+  ├──<digest_of_signature_manifest>
+    ├──<signing algorithm>
+    ├──<signed attributes>
+    ├──<unsigned attributes>
+    ├──<certificates>
+    └──<payload>
+```
+
+## Outline
+
+```text
+Inspect all signatures with the signed artifact.
+
+Usage:
+    notation inspect [flags] <reference>
+  
+Flags:
+   -h, --help for describing the signature
+   -o, --output on command line sets the output to json
+   -p, --password string password for registry operations (default to $NOTATION_PASSWORD if not specified)
+       --plain-http registry access via plain HTTP
+   -u, --username string username for registry operations (default to $NOTATION_USERNAME if not specified)
+```
+
+## Usage
+
+### Display the details of all the listed signatures and its associated certificate properties of the signed container image
+
+
+```text
+notation inspect [flags] <registry>/<repository>@<digest>
+```
+
+## Inspect signatures on the supplied OCI artifact identified by the digest
+
+```shell
+# Prerequisites: Signatures are stored in a registry referencing the signed OCI artifact
+notation inspect localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
+```
+
+An example output:
+```shell
+localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
+└──application/vnd.cncf.notary.signature
+    ├── sha256:aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
+      ├──signing algorithm: RSASSA-PSS-SHA-256
+      ├──signed attributes:
+            ├──contentType: application/vnd.cncf.notary.payload.v1+json
+            ├──signingScheme: notary.default.x509
+            ├──signingTime: 2022-04-06T07:01:20Z
+            ├──expiry: 2022-10-06T07:01:20Z
+            ├──verification plugin: com.example.nv2plugin                   //extended attributes to support plugins
+      ├──user-defined attributes:
```

Comment:
Are the user-defined attributes signed or not?

Consider `system-defined` attributes and `user-defined` attributes.